app/src/main/java/com/github/se/travelpouch/model/TravelContainer.kt

```diff
@@ -0,0 +1,93 @@
+package com.github.se.travelpouch.model
+
+import com.google.firebase.Timestamp
+
+/**
+ * Data class representing a travel container.
+ *
+ * @property fsUid Firestore UID of the travel container.
+ * @property title Title of the travel.
+ * @property description Description of the travel container, can be a blank string.
+ * @property startTime Start time of the travel.
+ * @property endTime End time of the travel.
+ * @property location Location details of the travel.
+ * @property allAttachments Map of attachments, name of the attachment and Firestore UID of the
+ *   attachment.
+ * @property allParticipants Map of participants, Participant and their Role. One owner must exist
+ *   at all times unless the travel container is deleted.
+ */
+data class TravelContainer(
+    val fsUid: String, // Firestore UID
+    val title: String,
+    val description: String, // can be a blank string
+    val startTime: Timestamp,
+    val endTime: Timestamp,
+    val location: Location,
+    val allAttachments:
+        Map<
+            String,
+            String>, // list of attachments, key is the name of the attachment, value is the uid of
+    // the attachment
+    val allParticipants:
+        Map<
+            Participant,
+            Role>, // One owner must exist at all times unless travel container is deleted
```

Comment:
Consider using immutable maps for allAttachments and allParticipants. This promotes safer and more predictable state management. If you need to update these collections, you can create a new instance with the updated state.

In the view mode (that will be created soon later in the project) we can add a function like the one below to ensure more secure state management
```kotlin
fun addParticipant(newParticipant: Participant, role: Role) {
    val updatedParticipants = allParticipants.toMutableMap()
    updatedParticipants[newParticipant] = role
    // Create a new TravelContainer with the updated participants
    travelContainer = travelContainer.copy(allParticipants = updatedParticipants.toMap())
}
```

Reply:
Hi, if I'm not mistaken the Map interface is immutable by default in Kotlin. Do not worry, I share you view on this matter.
![Sun Oct  6 08:49:38 PM CEST 2024select](https://github.com/user-attachments/assets/b53d9050-41ed-47fe-8abb-f32ae1a4e280)